lib/util/source-code-fixer.js

```diff
@@ -88,32 +125,23 @@ SourceCodeFixer.applyFixes = function(sourceCode, messages) {
 
     if (fixes.length) {
         debug("Found fixes to apply");
+        let fixesWereApplied = false;
 
         for (const problem of fixes.sort(compareMessagesByFixRange)) {
-            const fix = problem.fix;
-            const start = fix.range[0];
-            const end = fix.range[1];
-
-            // Remain it as a problem if it's overlapped or it's a negative range
-            if (lastPos >= start || start > end) {
-                remainingMessages.push(problem);
-                continue;
-            }
-
-            // Remove BOM.
-            if ((start < 0 && end >= 0) || (start === 0 && fix.text.startsWith(BOM))) {
-                output = "";
+            if (typeof shouldFix === "function") {
+                if (shouldFix(problem)) {
+                    fixesWereApplied = attemptFix(problem) || fixesWereApplied;
```

Comment:
If I'm understanding correctly, `attemptFix` will only return `false` if the current fix conflicts with another fix that has already been applied. In that case, will `fixesWereApplied` ever be `false` after this line?